spatialdata/_core/models.py

```diff
@@ -0,0 +1,407 @@
+        geo_df.attrs = {"transform": transform}
+        cls.validate(data)
+        return geo_df
+
+    @parse.register
+    @classmethod
+    def _(
+        cls,
+        data: GeoDataFrame,
+        transform: Optional[Any] = None,
+        **kwargs: Any,
+    ) -> GeoDataFrame:
+
+        if transform is None:
+            transform = Identity()
+        data.attrs = {"transform": transform}
+        cls.validate(data)
+        return data
+
+
+class ShapeModel:
+    coords_key = "spatial"
+    transform_key = "transform"
+    attrs_key = "spatialdata_attrs"
+
+    def validate(self, data: AnnData) -> None:
+        if self.coords_key not in data.obsm:
+            raise ValueError(f"AnnData does not contain shapes coordinates in `adata.obsm['{self.coords_key}']`.")
+        if self.transform_key not in data.uns:
+            raise ValueError(f"AnnData does not contain `{self.transform_key}`.")
+        if self.attrs_key not in data.uns:
+            raise ValueError(f"AnnData does not contain `{self.attrs_key}`.")
+        if "type" not in data.uns[self.attrs_key]:
+            raise ValueError(f"AnnData does not contain `{self.attrs_key}['type']`.")
+        if "size" not in data.uns[self.attrs_key]:
+            raise ValueError(f"AnnData does not contain `{self.attrs_key}['size']`.")
+
+    @classmethod
+    def parse(
+        cls,
+        coords: np.ndarray,  # type: ignore[type-arg]
+        shape_type: Literal["Circle", "Square"],
+        shape_size: np.float_,
+        transform: Optional[Any] = None,
+        **kwargs: Any,
+    ) -> AnnData:
+        """
+        Parse shape data into SpatialData.
+
+        Parameters
+        ----------
+        coords
+            Coordinates of shapes.
+        shape_type
+            Type of shape.
+        shape_size
+            Size of shape.
+        transform
+            Transform of shape.
+        kwargs
+            Additional arguments for shapes.
+
+        Returns
+        -------
+        :class:`anndata.AnnData` formatted for shapes elements.
+        """
+
+        adata = AnnData(
+            None,
+            obs=pd.DataFrame(index=np.arange(coords.shape[0])),
+            var=pd.DataFrame(index=np.arange(1)),
+            **kwargs,
+        )
+        adata.obsm[cls.coords_key] = coords
+        if transform is None:
+            transform = Identity()
+        adata.uns[cls.transform_key] = transform
+        adata.uns[cls.attrs_key] = {"type": shape_type, "size": shape_size}
+        return adata
+
+
+class PointModel:
+    coords_key = "spatial"
+    transform_key = "transform"
+
+    def validate(self, data: AnnData) -> None:
+        if self.coords_key not in data.obsm:
+            raise ValueError(f"AnnData does not contain points coordinates in `adata.obsm['{self.coords_key}']`.")
+        if self.transform_key not in data.uns:
+            raise ValueError(f"AnnData does not contain `{self.transform_key}`.")
+
+    @classmethod
+    def parse(
```

Comment:
Here there is the need for another array being passed to the parser, in fact since points can't be annotated by tables, we need to include the annotation when constructing them, which is one extra column specifying the cell type for each point.

Reply:
I have included this in the parser, gonna push it soon.